### ISSUE TITLE
Towne - typed line cuts off

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1097]
In GitLab by @kdjstudios on Sep 12, 2018, 10:07

**Submitted by:** Deb Crown <dcrown@towneanswering.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2018-09-12-65192/conversation

**Server:** External

**Client/Site:** Towne

**Account:** 3571

**Issue:** 

I noticed that a Staged Fee that I Programmed on Acct 3571 did not properly print on the invoice what I had typed.

It printed some of it and then did th…

 

If you look at the current invoice (TWN 45322) for that account, you will see what I am talking about.

 

I understand that SA has limits to what will print but here are my comments:
1. There is actual space on the invoice for the words to print but they don’t

2. Why does SA allow us to type if the words won’t print? If you are going to limit characters, we should know that.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 13, 2018, 06:53

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 13, 2018, 06:57

Hi [Kyle] ( @kdjstudios ),  

Please guide us from here that what should we do? 

Initially, our thoughts are to increase the limits of characters and should show the same info on Help section.

Please provide your input to proceed further.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 13, 2018, 08:21

@pchaudhary 

This will need to be discussed with @tim.traylor and Gary will need to approve any changes we make. 

My personal thoughts, would be to not increase what is being shown on the invoice, but rather limit what can be typed in the description field. I would anticipate this will be the most effective and efficient change. As having to update all of the different invoice formats can certainly cause other concerns and will most likely be far more complex.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 13, 2018, 10:22

@tim.traylor suggested that we make it a setting at the site level for them to determine to "force descriptions to one line". 

I wrote:

> Thank you for notifying us. We have informed the support team and are currently discussing how best to handle this. Once we have an update we will keep you informed.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Sep 25, 2018, 07:02

@kdjstudios 

We have worked on following  site and account page for `one line description` 


*********** Account Edit Page  ************

We have fixed the description of recurring fees and activities stages fee.

![image](/uploads/bedd45adc9f6d950d34ee802185980cc/image.png)

![image](/uploads/58f3b0d36e1c695710ada999df1b5aad/image.png)





*********** Site Edit Page  ************ 

we have fixed the site activities and 

![image](/uploads/115007da1dbe4d0b17f6bc3e4df256a8/image.png)


![image](/uploads/2069f48a2cf6ccc40f9f7687fb6c4fd7/image.png)



**When we add more the 80 characters in the description then we are presenting by the following validation error.**
 
![image](/uploads/535cc5a2b4f89de896a8b0395eee14f2/image.png)



>> Note :  This branch  is live for testing on merge2 server  till  tomorrow moring 3am EST

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 08:31

@amishra 

Thank you, but I believe we need to have a checkbox setting at the site level to turn this limiting on or off.

If turned on then the above would take effect; but rather then being a error, it just needs to be a warning. something like "Only the first 80 characters will be printed on the invoice". When printing the invoice it would insure that only the text takes up one line. 

if turned off then there would be no warnings at the account level and when printed to the invoice a charge may take multiple lines. 

@tim.traylor  any additional input or clarifications?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Sep 26, 2018, 07:05

@kdjstudios 

Total estimate is `1.5 hrs` including testing.

*  Development: 1hrs
*  Testing : 0.5 hrs

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 26, 2018, 10:30

@amishra 

Thanks for the update. lets please discuss this in a meeting with @tim.traylor before proceeding with any work.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 12, 2018, 12:04

@amishra @pchaudhary @smasih @tim.traylor 

Deb wrote:

> I sent this email a month ago…and I am sure I received a ticket #...but I can’t find it, so I apologize for not using it.
> 
> Does anyone have an answer for my question?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 12, 2018, 12:06

I wrote:

> Thank you for following up. The support team has been discussing different proposed solutions. So far there is no update we have yet to provide you. Once there is an update we will inform you.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Oct 15, 2018, 07:09

@amishra Hi Ashish, I'm on board with the change - it's only 1.5 hours, so please go ahead and include it in the next release.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 15, 2018, 09:42

I wrote:

> We just wanted to send you an update. The team was able to find the best possible solution to insure that the you will have the option to allow for multi line charge labels on invoices. We believe we will have this be a setting (Check Box) at the site level to allow for limiting the label text to one line on an invoice. By enabling the check box it will insure that all line items on an invoice only take up one line. This will be recommended if you want your invoices to maintain on one page. While if this is disabled, then the full description text will be printed on the invoice items and may result in an invoice being spread across multiple pages. Currently it is being developed and we do not have an ETA on when it will be released to production. We will keep you informed once it is released.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 26, 2018, 15:13

@amishra @pchaudhary @smasih @tim.traylor 

I am just reaching out for an update, As the last comments on this state this should have been ready for this release.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Oct 30, 2018, 07:54

@kdjstudios 

We have started working on it and once it completed I will let you know.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Oct 30, 2018, 07:54

added 30m of time spent at 2018-10-30

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Oct 31, 2018, 07:26

@kdjstudios 

We have completed the functionality of limiting characters for account's
`Recurring Fees`, `Staged Fees` and `Activities`. Please test it and let me know if there are any changes.
 
**Test cases and results:**

Case 1. `when we turn on the limit in site edit page.` (**it's checked by default**)
    ![image](/uploads/6615136641082e56f6de47a007230b18/image.png)  
  * In account show page we have only shown 80 characters in the description fields.
    ![image](/uploads/d282b7ebfaad9c1ddfd64a25536ebc27/image.png)
  * In account edit page we have shown the warning when we input the 80 characters above.
    ![image](/uploads/6a02a0a8e683651f55427989326b18cd/image.png)
  * Draft invoice.
    ![image](/uploads/68aeb001c84d8bddab1fe4ed9b056cae/image.png)
  * Final Invoice
    ![image](/uploads/a0913575e557ba5c4c30a23d8b21457d/image.png)

Case 2. `When we turn off the limitation on the description from the site edit page under the settings section.`
     ![image](/uploads/da911a4808d7b346a7c525eef9513278/image.png)
   * In account show page.
      ![image](/uploads/243f01ce0c485c4b8169a963a8f619bf/image.png)
   * In account edit page there is no warning shown.
      ![image](/uploads/3ed36503316b8097dc0918f83f37c104/image.png)
   * Draft invoice   
      ![image](/uploads/431373ca0ad3fadab61823d823bccb90/image.png)
   * Final Invoice
     ![image](/uploads/52a87abd7e689bb147aa8a6e50cdaf25/image.png)


I'll let you know on which server this functionality is live by my end of the day. 

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 31, 2018, 09:08

@smasih @pchaudhary @amishra @tim.traylor 

Thank you for the update. Just from looking at the screen shots, it looks good. But yes please let me know which server this is on and when it is ready for my testing. Thank you.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 9, 2018, 07:26

@kdjstudios 

We have set up for testing in the merge2 server. Please test it.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 9, 2018, 09:46

Hello Team, 

## Summary

I used the below text for all inputs. The following was the result of each functionality. 

> Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum. 

In conclusion from all of the testing and results pointed out below I found the following:
- Staged fees seem to have their own character limit and do not work with the setting.
- Activities and Additional Fees at the Site level do not work with this setting.
- When Enabled, and re-editing an account/site with this enabled, I believe we should display the warning then if possible too. 
- I know we are going to be releasing another invoice format that has the allowance, usage, and overage added to this line of text. What will we need to do to insure the character limit functions in a similar manor for that too. (I believe we will need to reduce the char limit even more to allow for the additional text to be displayed correct?)
- Everything else worked as expected. 

## Staged Fees:

**Enabled:**

Warning displayed correctly and only this was saved "Lorem Ipsum is simply dummy...". Also when printed on the invoice it matched this exactly. 
![image](/uploads/1f424fee962824cfa1aae38e80c16e50/image.png)

When adding a staged fee with a long name at the site level, No warning was displayed, but it did reduce the text when saved. to the same as above.  

**Disabled:**

When saving the staged fee the text was limited to "Lorem Ipsum is simply dummy...".
![image](/uploads/af43523b57616a74e3743871825ad596/image.png)

## Account Edit Recurring Fees:

**Enabled:**

Warning displayed correctly when entering; but allowed for the full text to be saved. When viewing the account again to edit, warning was not displaying.  When printed on invoice, it limited the text to "".
![image](/uploads/a4d3579673b88e4088164623b6d5c8d5/image.png)

**Disabled:**

Allowed for all the text above and when printed on the invoice it matched this exactly. 

## Account Edit Activities:

**Enabled:**

Warning displayed correctly when entering; but allowed for the full text to be saved. When viewing the account again to edit, warning was not displaying.  When printed on invoice, it limited the text to "".
![image](/uploads/79e740858160e43f8d43abbc0cb82b0f/image.png)

When trying to add the activity with the large name at the site level, it almost messed up the drop down.
![image](/uploads/c4f414f3d8dab455397021487e374e08/image.png)

**Disabled:**

Allowed for all the text above and when printed on the invoice it matched this exactly. 


## Account Info Screen:

**Enabled:**

Only displayed "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lo..." for both the Recurring Fees and Activities. 
![image](/uploads/2a46984fa2d277f95b4f63420da05233/image.png)

When adding a activity with a long name at the site level, it appears here in full. 
![image](/uploads/45c458303626641fd6e1c26e36bfe9d5/image.png)

**Disabled:**

Allowed for all the text above and displayed 

## Draft Invoice viewing:

**Enabled:**

Staged Fee, Activity, and Recurring all displayed as expected.
![image](/uploads/3add9cf3a65e003f696db125caddb0ec/image.png)

**Disabled:**

Allowed for all the text above and when printed on the invoice it matched this exactly. NOTE: this excludes the staged fee issue outlined above.   

## Draft Invoice Additional Fees:

For some reason I thought when adding an additional fee on the draft invoice screen we could adjust the description. but It appears that is no longer the case. It only allows for the name at the site level.
![image](/uploads/0c0523335d97897e8292ba19ba6f6a69/image.png)

## Final Invoice:

**Enabled:**

Everything appeared as shown on the draft invoice screen. 
![image](/uploads/f7fc774b5c9750886e365e25175c2999/image.png)

**Disabled:**

Allowed for all the text above and when printed on the invoice it matched this exactly. 

## Site Edit Additional Fees:

**Enabled:**

No warning and allowed for the full text to be saved in the Name field. 

**Disabled:**

Allowed for all the text above and when printed on the invoice it matched this exactly. 

## Site Edit Activities:

**Enabled:**

No warning and allowed for the full text to be saved in the Name and Description Fields.

**Disabled:**

Allowed for all the text above and when printed on the invoice it matched this exactly.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 10, 2018, 04:27

@kdjstudios 

We have fixed all the issues mentioned above.


> Activities and Additional Fees at the Site level do not work with this setting.

On the invoice page, the description comes from account, not from the site so its not required. Should we add on the site level as well? please let us know.


> I know we are going to be releasing another invoice format that has the allowance, usage, and overage added to this line of text. What will we need to do to insure the character limit functions in a similar manor for that too. (I believe we will need to reduce the char limit even more to allow for the additional text to be displayed correct?)

Yes right, so we have reduced the character limit to 50 from 80.

**If all seems good from your end please push this to ready.**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 12, 2018, 11:53

@amishra @pchaudhary @smasih @tim.traylor 

Marking this for discussion. We will want to get @tim.traylor input on this. I personally feel we will need to have this function the same at the site level. This is due to two reasons: first on the draft invoice screen when selecting a fee it pulls it from the site level, not the account level; and second because on the account drop downs the text is so long it makes the drop down hard to use. 

In regards with the character limit, I believe that should work fine; but we will want to test it with the new invoice format to insure it works properly. 

In addition to this, I just want to confirm that all places are now using the same character limit. As previously the staged fee section operated differently.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 13, 2018, 05:26

@kdjstudios 

Thanks for your valuable suggestion and we understand your concern about the description pulling from the site level which needs more adjustment than current but we would like to inform you that there is a select box named as *Limit line items characters on invoice to 50 recommended.* which is used to set the functionality you're talking about.   
  ![image](/uploads/247c3ea0ebc3d2ed707b2aa6d4fa6daf/image.png)

If the select box is checked then the account staged fee, draft invoice and account edit pages will show only 30 characters in the drop-down and 50 in the description else there is no limitation on the characters.

So basically, we have given the choice to the user to limit the characters if he/she wants.


Currently, the branch is different on the server so you can not see the select box but we'll let you know once we will switch on the concern branch.  

Hope this helps.  
Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Nov 13, 2018, 10:36

@smasih @kdjstudios Recapping our discussion:
* We do not limit the length of the description fields in the UI.  They can enter as long a description as they like.
* We need a single indicator at the site level for 'Limit line item descriptions on invoices to a single line'.
* The invoice processing should check that setting and truncate the line items if requested.
* For activities, where we have a dynamic portion of the label (i.e., 10 allowed, 20 over, etc.) we want to shorten the static portion of the description so when it is added to the dynamic portion it does not exceed the character limit.  Again, that's only if the set the limitation flag.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 14, 2018, 06:04

Linking ticket #600 which includes the allowance, usage and billable quantities for the invoice line description.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 23, 2018, 07:22

@kdjstudios 

We have setup merge #600 and #1099 and ready for testing on the merge2 server. please test it and provide me the feedback.

thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 23, 2018, 11:30

@amishra @smasih @pchaudhary @tim.traylor 

Thank you, Everything except for the following appear to be working

- For activities, where we have a dynamic portion of the label (i.e., 10 allowed, 20 over, etc.) we want to shorten the static portion of the description so when it is added to the dynamic portion it does not exceed the character limit.  Again, that's only if the set the limitation flag.
- Taxes are displaying the full label/name/description and are not limited to one line.
- Holidays and Late charges are not displaying the description, but instead are just displaying "Late Charge" or "Holiday". I think there is already a ticket with this issue, I will link to it once I find it. (#458)

Here is an invoice from my testing to show the above: [invoice_AVZTST-239.pdf](/uploads/455ac7f8aa581a7bc068080c3976dfcb/invoice_AVZTST-239.pdf)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 26, 2018, 06:58

> For activities, where we have a dynamic portion of the label (i.e., 10 allowed, 20 over, etc.) we want to shorten the static portion of the description so when it is added to the dynamic portion it does not exceed the character limit.  Again, that's only if the set the limitation flag.

* We have fixed the description limit for activities. In the case of activity, if the flag of invoice line character limitation on Site level is True, then it's going to truncate the activity description such that the static + dynamic portion will remain on one line. The change is for only the PDF version not for the draft page.  
 

> Taxes are displaying the full label/name/description and are not limited to one line.

* We have fixed the Tax name with maximum 80 character and we are also showing the warning for the name label.



![image](/uploads/d412aff6ed33a0c67d05ba2f40d41bfa/image.png)

> Holidays and Late charges are not displaying the description, but instead are just displaying "Late Charge" or "Holiday". I think there is already a ticket with this issue, I will link to it once I find it. (#458)

* We will fixed this issue under the ticket. #458 

thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 26, 2018, 07:07

assigned to @amishra and unassigned @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 26, 2018, 07:07

@kdjstudios 

Please test it on the Merge2 server and update accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2018, 09:48

@pchaudhary 

Please be sure to upload the correct branch. There is no ability to specify to limit characters on the site edit page on MERGE2.

![image](/uploads/8e176935a769b981cdfb7f27a95cef83/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 28, 2018, 07:02

@kdjstudios 

Sorry, it's our bad.
We request you to please test it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 28, 2018, 10:40

@pchaudhary 

Thank you. I have the checkbox selected and when I went to test to insure the Tax title behaved correctly. It was cut off when viewing in the draft invoice screen, but after finalized the title printed out fully. 

![image](/uploads/164188f10dafbecfdc10a773a7f592a3/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 29, 2018, 06:31

@kdjstudios 

Please look into this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 29, 2018, 08:53

@pchaudhary @amishra @smasih @tim.traylor 

I am sorry, I think you may have mentioned the wrong user to look into this. As I am the one reporting it. 

![image](/uploads/0b90cbab3fa9de100b814b67892a1e9c/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 29, 2018, 21:24

Hi [Kyle] ( @kdjstudios ), 

Yes, sorry for that. There was a typo, I was supposed to mention @amishra on this.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 29, 2018, 22:59

@pchaudhary @amishra 

while understanding we want to test every aspect of it, but I don't think so someone uses or will use a huge text just for taxes so let's not spend more time on commenting and on this ticket and just put a 80 char limit on the creation of taxes/late fees/holiday *(if it's not already there)* and that should be it. @kdjstudios let me know if you agree.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 30, 2018, 08:56

@pchaudhary @amishra @amishra @tim.traylor 

As stated by @tim.traylor the Holidays, Taxes, Fees, etc should all behave the same when being printed on an invoice. Please insure this is setup per the request.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Dec 3, 2018, 02:50

@kdjstudios 

Let's keep Holidays and Late Charge aside as currently, system is not printing their description/name. We can pick holidays and LC once the other ticket(#458) related to it is completed.

For taxes, the UI on account level is different. It's just a select box on account level so the character limit/warning has to be on site level/site edit page. So, we'll imitate the same behaviour.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 3, 2018, 10:50

@smasih 

I agreed lets keep the Holiday and Late charges issues in the #458 ticket.

In regards to the Taxes, I just want to clarify your statement, that as requested by Tim, when a user has selected the "Limit" option on the site level, the Tax length when printed on an invoice will be limited also? Because it would appear that @pchaudhary had previously developed this but it was not complete. It would limit the length on the draft invoice page, but not when printed on the final invoice. As I pointed out in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1099#note_30804

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Dec 4, 2018, 06:26

@amishra @pchaudhary 

> ad previously developed this but it was not complete. It would limit the length on the draft invoice page, but not when printed on the final invoice. As I pointed out in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1099#note_30804

please finish the development on this as there is not more left on this ticket.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 21, 2018, 12:11

@smasih @amishra @pchaudhary @tim.traylor 

Where are we at with development and testing on this? We probably will want to release this at the same time we release #1268

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 24, 2018, 10:46

Updating priority on the ticket, as we now have a second client complaint about this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 24, 2018, 10:47

@smasih @amishra @pchaudhary @tim.traylor 

What is the status of this? Is this ready for testing? I am putting back into Dev until we have an update.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 26, 2018, 01:15

@kdjstudios 

According to @smasih comment http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1099#note_30981 we have added same functionality on taxes. 
 
We have setup on the dev server for testing. please test it.

**Testing Step**
http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1099#note_29214 as well taxes too.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 26, 2018, 09:37

@amishra @smasih 

Thank you for the update. A quick check confirmed the char limit functionality is there and appears to be working. However, we will want to test this in conjunction with usage invoice formatting in #1268. It looks like #1268 should be on this same server, but was probably on a different branch. 

May we please have both of those functionalities on a testing server (same branch) to see if everything works together?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 27, 2018, 24:24

@kdjstudios 

we have set up both branch #1099 and #1268 on the same dev server. please test it.  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 27, 2018, 09:17

@amishra 

Thank you. It appears that the Ability to limit to one line is working correctly in conjunction with the Usage description. Marking this for Announcement.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 5, 2019, 04:34

@kdjstudios 

While testing on the production server we found another scenario on which it is failing at the moment due to which we did not release it today. We will update you once fixed it.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 6, 2019, 23:25

@kdjstudios 

We have fixed the issue and please test it on dev server.

Issue: 

when adding long text in the tax name inside the site and select it on the account. Then not showing tax name rows/columns correctly. This is a new bug but we have fixed it.

**Before**
![image](/uploads/77c7ce04bf849d350fb30c30ce2ba88b/image.png)


**After**
![image](/uploads/95a85d59ebb0335960e64d2eb51941a3/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 7, 2019, 08:29

@amishra 

Why is there a huge blank space to the left? It seems the draft invoice screen now has a very uneven layout. Please insure that all text is aligned with the rest of the descriptions. Also insure the description columns is wider then the other columns; as currently base rate and quantity columns have too much space. 


Incorrect column layout:

![image](/uploads/1781d37ddf89e123411fea02c0092f60/image.png)

How the column layout should look:

![image](/uploads/18a2a00aed963c2775ceff3fab738b1b/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 8, 2019, 02:47

@kdjstudios 


We would like to inform you that all the columns are floating type that means it depends on description text if description text less text then all columns are moving left as well right.


And taxes text we have fixed to description columns.

Please test it on the dev server.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 8, 2019, 02:48

added 1h of time spent at 2019-01-08

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 8, 2019, 10:06

This was discussed today and @smasih stated he reviewed that this needs to display like it was before with @amishra and that it was completed. I just tested on Dev and all appears to be in a nice layout. Marking for Announcement.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 8, 2019, 10:12

@amishra @pchaudhary @smasih 

I was reviewing the finalized invoice and it would appear that the double spacing is occurring. I just want to make sure we do not end up with it once released to production like with #1304

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 9, 2019, 04:42

@kdjstudios 

We have setup latest code on this branch and we are sending our testing screenshot with all cases. Please have a look.

**Case 1.**  When  limit line item descriptions on invoices to a single line **and** Usage Display Format On Invoice both are **true**
![Screenshot_from_2019-01-09_17-02-11](/uploads/c437cc7e94a456e0b4270a662d8c737f/Screenshot_from_2019-01-09_17-02-11.png)

**Case 2.** When limit line item descriptions on invoices to a single line **false** **and** Usage Display Format On Invoice are **true**.

![host_usages](/uploads/038bdf33c8aff4cee9efab8e4c540116/host_usages.png)

**Case 3.** When  limit line item descriptions on invoices to a single line **true** **and** Usage Display Format On Invoice are **false**.
![Screenshot_from_2019-01-09_17-04-46](/uploads/fa68bd33d4adea1e9b29403310a926c9/Screenshot_from_2019-01-09_17-04-46.png)

**Case 4.** When  limit line item descriptions on invoices to a single line **and** Usage Display Format On Invoice both are
 **false**.
![Screenshot_from_2019-01-09_16-57-30](/uploads/78786b054db5a48c2cd98a0a6a15676d/Screenshot_from_2019-01-09_16-57-30.png)


**As per current code We are hiding both negative and zero amount of the child accounts.**  
**So my question is If the line item account amount is negative then should we show?**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 9, 2019, 08:42

@amishra @pchaudhary @smasih @tim.traylor @rchauhan @abakshi 

**As per current code We are hiding both negative and zero amount of the child accounts.**  

> - When did we start hiding negative amounts? Also can you clarify, is this hiding negative line items, negative child accounts, or negative everything? I do not recall this ever being discussed or reviewed? Please reference a ticket where this was approved. 

> - We are only supposed to hide terminated child accounts with zero balances, is that what you mean to say? #1229 

**So my question is If the line item account amount is negative then should we show?**

 > - Please see above questions. I am wondering what negative amounts/accounts you are referring to and when this was approved.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 10, 2019, 04:00

@kdjstudios 

* If child accounts are active (show in any case)
* If child accounts are inactive and the sum total is **$0**.(Do not show)
* If child accounts are inactive and the sum total is greater than **$0**. (show)
* If child accounts are inactive and the sum total is less than **$0**. (show)

Currently, these steps are follow on production.
Please confirm this step is right. 

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 10, 2019, 08:36

@amishra 

Thank you for the clarification. Yes that is correct. Marking this as ready, because it was already announced.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 19, 2019, 04:41

@kdjstudios 

Releases on both of the production server.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 19, 2019, 21:59

@pchaudhary 

Thank you for the update. May we please have this disabled for all site by default. It looks like it was set to be enabled for all the sites I checked. But we want the users to enable this at their choosing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 19, 2019, 22:02

I wrote:

> We are just following up. The support team was able to release an update to SA Billing that now has the ability to choose if you would like the invoice line items to display the full label description or to limit the label to only one line. By enabling this option you will insure that each line item only uses one line and can lead to a simplified bill that will usually fit on one page, depending upon how many billable items you have on the invoice. If this is disabled the full description on the line item will be displayed on the invoice and can lead to a more descriptive invoicing.  This setting is found within the ‘Site Edit’ page in the ‘Settings’ section and is labeled ‘Limit line item descriptions on invoices to a single line’. Please note this is disabled by default.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 21, 2019, 08:54

@pchaudhary 

Marking this back for development. Please insure that this setting is disabled by default for all sites internal and external. Once that is completed please update label to ops/client to mark done.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 24, 2019, 24:51

@kdjstudios 

> I am confused with how we determined a site checked that box? can you please provide the username and date time those were checked? I want to make sure we are covered in case the client or sites complain.  

* We can not determine it that who & when updated this as we are not locking this details anywhere in SAB. Please confirm with the sites or they can change it on their own.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 21, 2019, 23:18

@kdjstudios 

We have disabled this setting on the backend by default for all of the sites on both of the production servers.  

Still, these sites having this option as checked.  
  * Memphis, TN
  * Orlando, FL
  * Portland, OR
  * Keener Communications
  * Laser Answering Service

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 22, 2019, 11:28

@pchaudhary 

Why did we not disable the "Limit line item descriptions on invoices to a single line" setting on the back end for those sites?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 22, 2019, 23:03

@kdjstudios 

> Why did we not disable the "Limit line item descriptions on invoices to a single line" setting on the back end for those sites?

* We have changed the default settings & would like to inform you that these sites checked those fields from them, because of which it's not false.

Hope this helps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 23, 2019, 08:47

@pchaudhary 

I am confused with how we determined a site checked that box? can you please provide the username and date time those were checked? I want to make sure we our covered in case the client or sites complain. 

NOTE: When released those boxes were checked by default for all sites and clients originally.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 24, 2019, 10:38

@pchaudhary 

This makes no sense once again. How can we determine that this was checked by the site, but we cannot determine when it was checked or by who? Please explain. 

Marking for review with @tim.traylor @smasih

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 25, 2019, 04:40

@kdjstudios 

I understand your concern & want to inform you that we are not logging every action of a user, from how we can provide you the name & details of the user.

I hope this helps.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 25, 2019, 09:28

@pchaudhary 

So how are we logging that these sites had disabled and then re-enabled their setting?

If you simply stating that a user made any change at the site level and then saved it, that does not mean that purposely enabled this setting and we need to have it disabled by default.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 27, 2019, 22:24

> So how are we logging that these sites had disabled and then re-enabled their setting?

@pchaudhary 

was this mentioned in the original requirements?



> If you simply stating that a user made any change at the site level and then saved it, that does not mean that purposely enabled this setting and we need to have it disabled by default.

@pchaudhary 
would you mind testing it out on any of the test server and put an end to the discussion, please.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 28, 2019, 02:45

Hi [Kyle] ( @kdjstudios ) & [Sumeet] (@smasih ),   

We understand that this topic is getting more time & energy only because of a small pop.  

* I personally went through the whole discussion above & found that [Ashish] (@amishra ) had already mentioned initially that we are making it **true by-default** on his (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1099#note_29214).     
*  **This whole discussion can be avoided if it was pointed out that time.**  

> So how are we logging that these sites had disabled and then re-enabled their setting?

* We are not logging that these sites had disabled & then re-enabled their setting.  
* This setting was true by default initially when developed.  
* Kyle pointed out that this should be false by default.  
* We had changed the by default setting as false.  
* But till then if any of the sites was updated in any case then that setting was set as true.  
* And here the above-mentioned sites was updated in between that time span because of which that setting is still true after changing the by default settings.  


* Now please let us know what should be done, so that we can close this ASAP.  


Hope this helps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 24, 2019, 23:01

Hi [Kyle] ( @kdjstudios ),  

We request you to please update this so that we can mark it as close.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 25, 2019, 08:51

@pchaudhary @amishra @smasih 

This is marked for review with @tim.traylor once it has been discussed we will post an update accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 26, 2019, 08:13

Marking for close, we have not received any complaints.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 26, 2019, 08:13

closed
